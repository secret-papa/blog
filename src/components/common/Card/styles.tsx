import styled from "styled-components";

export const Root = styled.div`
  width: 320px;
  background-color: #ffffff;
  box-shadow: rgba(0, 0, 0, 0.04) 0px 4px 16px 0px;
  border-radius: 4px;
`;

export const Header = styled.div``;

export const CoverImageWrap = styled.div`
  height: 168px;
`;

export const CoverImage = styled.img`
  display: block;
  width: 100%;
  height: 100%;
  object-fit: cover;
`;

export const Body = styled.div`
  padding: 1rem;
`;

export const Title = styled.h4`
  font-size: 1rem;
  font-weight: bold;
  margin-bottom: 0.25rem;
  word-break: break-word;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
  color: var(--color-grey-700);
`;

export const Description = styled.p`
  display: -webkit-box;
  height: 3.9375rem;
  font-size: 0.875rem;
  line-height: 1.5;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-line-clamp: 3;
  -webkit-box-orient: vertical;
  overflow: hidden;
  text-overflow: ellipsis;
  color: var(--color-grey-600);
  margin-bottom: 1.5rem;
`;

export const Meta = styled.div`
  font-size: 0.75rem;
  color: var(--color-grey-500);
`;
