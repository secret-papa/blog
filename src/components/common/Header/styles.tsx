import styled from "styled-components";

export const Wrapper = styled.header`
  display: flex;
  align-items: center;
  justify-content: center;
  height: 64px;
`;

export const Header = styled.div`
  display: flex;
  width: 1024px;
`;
