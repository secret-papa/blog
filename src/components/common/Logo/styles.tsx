import styled from "styled-components";

// eslint-disable-next-line import/prefer-default-export
export const Text = styled.span`
  font-size: 1.4rem;
  color: #2e4f4f;
`;
