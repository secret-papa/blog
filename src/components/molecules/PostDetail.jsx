import React from "react";
import PropTypes from "prop-types";
import { css } from "@emotion/react";
import { defineCustomElements as deckDeckGoHighlightElement } from "@deckdeckgo/highlight-code/dist/loader";
import "github-markdown-css";

import { Text } from "../atoms";

deckDeckGoHighlightElement();

const useStyles = () => ({
  root: ({ spacing }) => css`
    margin: ${spacing(5)}rem 0;
  `,
  header: ({ spacing }) => css`
    margin-bottom: ${spacing(4)}rem;
  `,
  title: ({ spacing }) => css`
    margin-bottom: ${spacing()}rem;
  `,
  body: () => css`
    ul {
      list-style: disc;
    }

    ol {
      list-style: decimal;
    }

    p {
      font-size: 1.8rem;
      line-height: 1.76;
    }

    blockquote {
      border-left: 0.25em solid #6bb0fa;
    }
  `,
});

const PostDetail = ({ title, date, content }) => {
  const styles = useStyles();

  return (
    <div css={styles.root}>
      <div css={styles.header}>
        <Text bold color="darkGray" component="h1" css={styles.title} variant="title1">
          {title}
        </Text>
        <Text color="lightGray" variant="body2">
          {date}
        </Text>
      </div>
      <div
        css={styles.body}
        className="markdown-body"
        // eslint-disable-next-line react/no-danger
        dangerouslySetInnerHTML={{ __html: content }}
      />
    </div>
  );
};

PostDetail.defaultProps = {
  date: "",
};

PostDetail.propTypes = {
  title: PropTypes.string.isRequired,
  date: PropTypes.string,
  content: PropTypes.string.isRequired,
};

export default PostDetail;
