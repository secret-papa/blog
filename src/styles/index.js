export { default as resetStyle } from "./reset";
export { default as theme } from "./theme";
