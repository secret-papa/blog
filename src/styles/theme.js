export default {
  colors: {
    grey: {
      100: "#F5F5F5",
      200: "#EEEEEE",
      300: "#E0E0E0",
      400: "#BDBDBD",
      50: "#FAFAFA",
      500: "#9E9E9E",
      600: "#757575",
      700: "#616161",
      800: "#424242",
      900: "#212121",
    },
  },
  font: {
    title1: {
      fontSize: "3.2rem",
      lineHeight: "1",
    },
    title3: {
      fontSize: "2rem",
      lineHeight: "1",
    },
    body1: {
      fontSize: "1.6rem",
      lineHeight: "1",
    },
    body2: {
      fontSize: "1.4rem",
      lineHeight: "1.4",
    },
    body3: {
      fontSize: "1.2rem",
      lineHeight: "1.4",
    },
  },
  spacing: (value = 1) => (value * 8) / 10,
};
