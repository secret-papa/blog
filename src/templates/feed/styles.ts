import styled from "styled-components";

export const Root = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  min-height: 100vh;
  background-color: #f8f9fa;
`;

export const Body = styled.main`
  width: 1024px;
`;

export const List = styled.ul`
  display: flex;
  flex-wrap: wrap;
  gap: 2rem;
`;
