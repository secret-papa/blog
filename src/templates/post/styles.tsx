import styled, { css } from "styled-components";
import { Link } from "gatsby";

export const Root = styled.div`
  margin-bottom: 3rem;
`;

export const Title = styled.h1`
  font-size: 3rem;
  font-weight: bold;
  color: var(--color-grey-900);
`;

export const Body = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  margin-top: 5.5rem;
`;

export const Article = styled.article`
  width: 768px;
`;

export const CoverImageWrapper = styled.div`
  margin-top: 2rem;
`;

export const CoverImage = styled.img`
  display: block;
  width: 100%;
`;

export const MarkdownWrapper = styled.div`
  margin: 5rem 0 6rem;
`;

export const MarkdownBody = styled.div`
  & ol,
  ul {
    list-style: decimal;
  }
`;

export const NavigationButtonWrapper = styled.div`
  display: flex;
  justify-content: space-between;
  width: 100%;
  gap: 1rem;
`;

const leftDirectionCss = css`
  text-align: left;
`;
const rightDirectionCss = css`
  text-align: right;
`;

type Align = "left" | "right";

export const NavigationButton = styled(Link)<{ align?: Align }>`
  display: flex;
  flex-direction: column;
  justify-content: center;
  gap: 0.5rem;
  width: 100%;
  height: 4rem;
  background-color: #f8f9fa;
  padding-left: 1rem;
  padding-right: 1rem;
  box-shadow: rgba(0, 0, 0, 0.06) 0px 0px 4px 0px;
  border: none;
  cursor: pointer;
  ${props => (props.align === "right" ? rightDirectionCss : leftDirectionCss)}
`;

export const NavigationButtonCaption = styled.span`
  font-size: 0.75rem;
  color: var(--color-grey-600);
`;

export const NavigationButtonTitle = styled.h3`
  font-size: 1.125rem;
  font-weight: bold;
  color: var(--color-grey-700);
`;
