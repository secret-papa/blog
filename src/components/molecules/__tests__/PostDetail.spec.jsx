import React from "react";

import { renderWithTheme } from "@tests";
import { theme } from "@src/styles";
import PostDetail from "../PostDetail";

describe("<PostDetail />", () => {
  it("should show correctly", () => {
    const { container } = renderWithTheme(
      <PostDetail title="title" date="0000-00-00" content="<p>content</p>" />,
      theme,
    );
    expect(container).toMatchSnapshot();
  });
});
