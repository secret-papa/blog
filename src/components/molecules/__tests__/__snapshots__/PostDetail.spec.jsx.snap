// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostDetail /> should show correctly 1`] = `
.emotion-0 {
  margin-bottom: 3.2rem;
}

.emotion-1 {
  color: #212121;
  font: bold 3.2rem/1 sans-serif;
  font-family: inherit;
  margin-bottom: 0.8rem;
}

.emotion-2 {
  color: #BDBDBD;
  font: normal 1.4rem/1.4 sans-serif;
  font-family: inherit;
}

.emotion-3 ul {
  list-style: disc;
}

.emotion-3 ol {
  list-style: decimal;
}

.emotion-3 p {
  font-size: 1.8rem;
  line-height: 1.76;
}

.emotion-3 blockquote {
  border-left: 0.25em solid #6bb0fa;
}

<div>
  <div>
    <div
      class="emotion-0"
    >
      <h1
        class="emotion-1"
      >
        title
      </h1>
      <span
        class=" emotion-2"
      >
        0000-00-00
      </span>
    </div>
    <div
      class="markdown-body emotion-3"
    >
      <p>
        content
      </p>
    </div>
  </div>
</div>
`;
